hyperspy/misc/eels/tools.py

```diff
@@ -307,3 +308,61 @@ def eels_constant(s, zlp, t):
         data=(t * i0 / (332.5 * ke)) * np.log(1 + (beta * tgt / eaxis) ** 2))
     k.metadata.General.title = "EELS proportionality constant K"
     return k
+
+
+def get_edge_onset(data, start, end, x_axis, percent_position):
+    """Get the onset energy for a core loss edge.
+
+    Parameters
+    ----------
+    data : NumPy array
+    x_axis : NumPy array
+        Same size as data
+    start, end : scalar
+        Region where the edge onset will be found, in x_axis values.
+    percent_position : float
+        At what fraction of the ELNES signal the onset energy will be
+        found. See EELSModel.estimate_and_set_coreloss_edge_onset for more
+        information.
+
+    Returns
+    -------
+    onset_energy : scalar
+
+    Examples
+    --------
+    >>> from hyperspy.misc.eels.tools import get_edge_onset
+    >>> s = hs.datasets.artificial_data.get_core_loss_eels_signal()
+    >>> data, x_axis = s.data, s.axes_manager[0].axis
+    >>> onset = get_edge_onset(data, 600, 700, x_axis, 0.1)
+
+    """
+    if not ((percent_position >= 0.) and (percent_position <= 1.)):
+        raise ValueError(
+                "percent position ({0}) must be between 0 and 1".format(
+                    percent_position))
+    start_i = np.argmax(x_axis > start)
+    end_i = np.argmax(x_axis > end)
+    if end_i == 0:
+        end_i = len(x_axis)
+
+    data = data[start_i:end_i]
+    x_axis = x_axis[start_i:end_i]
+
+    data_max = data.max()
+    data_min = data.min()
+    data_max_i = data.argmax()
+    data_min_i = data.argmin()
+    data_onset_value = (data_max - data_min) * percent_position + data_min
+
+    data_crop = data[data_min_i:data_max_i][::-1]
+    data_onset_i = np.argmax(data_crop < data_onset_value)
+    data_onset_i = len(data_crop) - data_onset_i
+
+    onset_slice = slice(data_min_i + data_onset_i - 1,
+                        data_min_i + data_onset_i + 1)
+
+    spline = InterpolatedUnivariateSpline(
```

Comment:
I still think that spline interpolation is an overkill feature here that breaks the "vectority" of the algorithm. Actually, here a spline is used just for linear interpolation. Why not writing the linear interpolator vectorially instead?